src/ImageProcessor.Web.Plugins.AzureBlobCache/AzureBlobCache.cs

```diff
@@ -378,24 +392,58 @@ public override void RewritePath(HttpContext context)
 
             if (this.streamCachedImage)
             {
+                // Map headers to enable 304s to pass through
+                if (context.Request.Headers["If-Modified-Since"] != null)
+                {
+                    request.IfModifiedSince = DateTime.Parse(context.Request.Headers["If-Modified-Since"]);
+                }
+
+                string[] mapRequestHeaders = { "Cache-Control", "If-None-Match" };
+                foreach (string h in mapRequestHeaders)
+                {
+                    if (context.Request.Headers[h] != null)
+                    {
+                        request.Headers.Add(h, context.Request.Headers[h]);
+                    }
+                }
+
                 // Write the blob storage directly to the stream
                 request.Method = "GET";
 
                 TryFiveTimes(
                     () =>
+                    {
+                        HttpWebResponse response;
+                        try
                         {
-                            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
+                            response = (HttpWebResponse)request.GetResponse();
+                        }
+                        catch (WebException ex)
+                        {
+                            // A 304 is not an error
```

Comment:
Can't believe I missed that. I need to check elsewhere in my codebase now for similar!